applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes_fractional.cpp

```diff
@@ -0,0 +1,2306 @@
+void TwoFluidNavierStokesFractional<TElementData>::CondenseEnrichment(
+    Matrix &rLeftHandSideMatrix,
+    VectorType &rRightHandSideVector,
+    const MatrixType &rHtot,
+    const MatrixType &rVtot,
+    MatrixType &rKeeTot,
+    const VectorType &rRHSeeTot)
+{
+    // Enrichment condensation (add to LHS and RHS the enrichment contributions)
+    double det;
+    MatrixType inverse_diag(NumNodes, NumNodes);
+    MathUtils<double>::InvertMatrix(rKeeTot, inverse_diag, det);
+
+    const Matrix tmp = prod(inverse_diag, rHtot);
+    noalias(rLeftHandSideMatrix) -= prod(rVtot, tmp);
+
+    const Vector tmp2 = prod(inverse_diag, rRHSeeTot);
+    noalias(rRightHandSideVector) -= prod(rVtot, tmp2);
+}
+
+template <class TElementData>
+void TwoFluidNavierStokesFractional<TElementData>::save(Serializer &rSerializer) const
+{
+    using BaseType = FluidElement<TElementData>;
+    KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, BaseType);
+}
+
+template <class TElementData>
+void TwoFluidNavierStokesFractional<TElementData>::load(Serializer &rSerializer)
+{
+    using BaseType = FluidElement<TElementData>;
+    KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, BaseType);
+}
+
+
+template <class TElementData>
+void TwoFluidNavierStokesFractional<TElementData>::CalculateOnIntegrationPoints(
+    const Variable<double> &rVariable,
+    std::vector<double> &rValues,
+    const ProcessInfo &rCurrentProcessInfo )
+{
+    if (rVariable == DIVERGENCE){
+
+        const auto& rGeom = this->GetGeometry();
+        const GeometryType::IntegrationPointsArrayType& IntegrationPoints = rGeom.IntegrationPoints(GeometryData::IntegrationMethod::GI_GAUSS_2);
+        const unsigned int num_gauss = IntegrationPoints.size();
+
+        if (rValues.size() != num_gauss){
+            rValues.resize(num_gauss);
+        }
+
+        Vector gauss_pts_jacobian_determinant = ZeroVector(num_gauss);
+        GeometryData::ShapeFunctionsGradientsType DN_DX;
+        rGeom.ShapeFunctionsIntegrationPointsGradients(DN_DX, gauss_pts_jacobian_determinant, GeometryData::IntegrationMethod::GI_GAUSS_2);
+
+        for (unsigned int i_gauss = 0; i_gauss < num_gauss; ++i_gauss){
+
+            const Matrix gp_DN_DX = DN_DX[i_gauss];
+            double DVi_DXi = 0.0;
+
+            for(unsigned int nnode = 0; nnode < NumNodes; ++nnode){
+
+                const array_1d<double,3> vel = rGeom[nnode].GetSolutionStepValue(VELOCITY);
+                for(unsigned int ndim = 0; ndim < Dim; ++ndim){
+                    DVi_DXi += gp_DN_DX(nnode, ndim) * vel[ndim];
+                }
+            }
+            rValues[i_gauss] = DVi_DXi;
+        }
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Class template instantiation
+
+template class TwoFluidNavierStokesFractional<TwoFluidNavierStokesFractionalData<2, 3>>;
+template class TwoFluidNavierStokesFractional<TwoFluidNavierStokesFractionalData<3, 4>>;
+
+} // namespace Kratos
```

Comment:
I made the suggestions in here, but please apply them to the generator and perform the automatic differentiation again.

Reply:
Sure,thanks!